src/app/hierarchical-grid/hierarchical-grid-validator-service-extended/hierarchical-grid-validator-service-extended.component.html

```diff
@@ -17,8 +17,12 @@
         </igx-column>
         <igx-column field="Fax" [editable]="true"></igx-column>
 
-        <igx-row-island [height]="null" [key]="'Orders'" [primaryKey]="'OrderID'" [autoGenerate]="false"
+        <igx-row-island [height]="null" #grid1 [key]="'Orders'" [primaryKey]="'OrderID'" [autoGenerate]="false"
             [batchEditing]="true" [rowEditable]="true" (formGroupCreated)="formCreateHandler($event)">
+            <igx-grid-toolbar [grid]="grid" *igxGridToolbar="let grid">
+                <button igxButton [disabled]="!grid.transactions.canUndo" (click)="undo(grid1)">Undo</button>
```

Comment:
You're passing the rowIsland, not the hierarchical grid intsance. Hence this will not work.
Please pass the grid itself, for example:
```
(click)="undo(grid)"
```